types/connect-mongo/index.d.ts

```diff
@@ -91,14 +91,19 @@ declare namespace connectMongo {
         mongooseConnection: mongoose.Connection;
     }
 
-    export interface NaitiveMongoOptions extends DefaultOptions {
+    export interface NativeMongoOptions extends DefaultOptions {
         db: mongodb.Db;
     }
 
+    export interface NativeMongoPromiseOptions extends DefaultOptions {
+        dbPromise: Promise<mongodb.Db>;
+    }
+
     export interface MongoStoreFactory {
         new (options: MongoUrlOptions): MongoStore;
         new (options: MogooseConnectionOptions): MongoStore;
-        new (options: NaitiveMongoOptions): MongoStore;
+        new (options: NativeMongoOptions): MongoStore;
+        new (options: NativeMongoPromiseOptions): MongoStore;
```

Comment:
This whole block should use a union type instead of four overloads:
```
new (options: MongoUrlOptions | MongooseConnectionOPtions | NativeMongoOptions | NativeMongoPromiseOptions): MongoStore
```

Reply:
Thanks @RyanCavanaugh, I've made the suggested changes.